From 77f5ca7c1ac549ac05cee58e848a81a77dae375e Mon Sep 17 00:00:00 2001
From: Suman Mukherjee <sumam@codeaurora.org>
Date: Mon, 12 Jan 2015 12:35:11 +0530
Subject: [PATCH 64/95] Camera: Added null check for memory reference

Null memory reference in releaseRecordingFrame will cause
crash in cfuzz tool.
Added Null check for memory reference

Change-Id: Id15fc55cc89b6039a36c1c3c7e4b55c0ae5aa296
---
 services/camera/libcameraservice/device1/CameraHardwareInterface.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/services/camera/libcameraservice/device1/CameraHardwareInterface.h b/services/camera/libcameraservice/device1/CameraHardwareInterface.h
index d983d35..e88bebb 100644
--- a/services/camera/libcameraservice/device1/CameraHardwareInterface.h
+++ b/services/camera/libcameraservice/device1/CameraHardwareInterface.h
@@ -327,6 +327,10 @@ public:
     void releaseRecordingFrame(const sp<IMemory>& mem)
     {
         ALOGV("%s(%s)", __FUNCTION__, mName.string());
+        if (mem == NULL) {
+             ALOGE("%s: NULL memory reference", __FUNCTION__);
+             return;
+        }
         if (mDevice->ops->release_recording_frame) {
             ssize_t offset;
             size_t size;
-- 
2.5.0

